include/circt/Dialect/SSP/Utilities.h

```diff
@@ -134,6 +169,27 @@ ProblemT loadProblem(InstanceOp instOp,
     prob.insertOperation(opOp);
     loadOperationProperties<ProblemT, OperationPropertyTs...>(
         prob, opOp, opOp.getPropertiesAttr());
+
+    // If set, ensure that the linked operator type has been loaded.
+    if (auto linkedOpr = opOp.getLinkedOperatorTypeAttr()) {
+      SymbolRefAttr oprRef = linkedOpr.getValue();
+      if (oprRef.isa<FlatSymbolRefAttr>())
+        // FlatSymbolRefs point to the instance's library; these operator types
+        // were already registered.
```

Comment:
should an assert be added here to verify this?

Reply:
👍 (The diff became a bit larger than expected because I found a missing check while thinking about the assert, and then tried to clarify what's happening while loading the operator types)